### PR TITLE
Acceptance test for sharing folders with group when receiver has matc…

### PR DESCRIPTION
**Pull request by bhawanaprasain**
## Description
It includes test for sharing folder with groups when receiver has a matching folder. 

## Related Issue
#35414

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

### CODE REVIEW

**Comment by bhawanaprasain** [comment]
@dpakach  @paurakhsharma @skshetry  please review on my PR.

**Comment by bhawanaprasain** [comment]
@dpakach @paurakhsharma @skshetry please review on my PR.

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 1]
```diff
@@ -254,4 +254,14 @@ Feature: Sharing files and folders with internal groups
...
       | Case-Sensitive-Group | CASE-SENSITIVE-GROUP | case-sensitive-group |
-      | CASE-SENSITIVE-GROUP | case-sensitive-group | Case-Sensitive-Group |
\ No newline at end of file
+      | CASE-SENSITIVE-GROUP | case-sensitive-group | Case-Sensitive-Group |
+
+    Scenario: share folder with a group when group has matching folder
+      Given user "user1" has logged in using the webUI
+      And the user has created folder "test"
+      When the user re-logs in as "user3" using the webUI
+      And the user creates a folder with the name "test" using the webUI
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: `test` folder creation for `user1` and `user3` can be done via API steps.

Comment (skshetry) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: Please use api steps to create folder.

**Comment by phil-davis** [comment]
@bhawanaprasain this probably also needs a rebase, due to PHPunit problems from last week that are now resolved.

**Comment by phil-davis** [comment]
@bhawanaprasain now there are conflicts. You are so lucky ;)

**Comment by bhawanaprasain** [comment]
@phil-davis I have resolved the conflicts.

**Comment by phil-davis** [comment]
@paurakhsharma @skshetry please review again.

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@bhawanaprasain please backport

**Comment by phil-davis** [comment]
Backport `stable10` #35850

**Pull request merged**